addons/metadata.integral.artists/lib/scraper.py

```diff
@@ -0,0 +1,355 @@
+                result[k] = v
+                if k == 'thumb':
+                    thumbs.append(v)
+                elif k == 'fanart':
+                    fanart.append(v)
+        if 'musicbrainz' in details:
+            for k, v in details['musicbrainz'].items():
+                result[k] = v
+        if 'fanarttv' in details:
+            for k, v in details['fanarttv'].items():
+                result[k] = v
+                if k == 'thumb':
+                    thumbs.append(v)
+                elif k == 'fanart':
+                    fanart.append(v)
+        # provide artwork from all scrapers for getthumb / getfanart option
+        if result:
+            # artworks from most accurate sources first
+            thumbs.reverse()
+            fanart.reverse()
+            thumbnails = []
+            fanarts = []
+            for thumblist in thumbs:
+                for item in thumblist:
+                    thumbnails.append(item)
+            result['thumb'] = thumbnails
+            for fanartlist in fanart:
+                for item in fanartlist:
+                    fanarts.append(item)
+            result['fanart'] = fanarts
+        data = self.user_prefs(details, result)
+        return data
+
+    def user_prefs(self, details, result):
+        # user preferences
+        lang = 'biography' + xbmcaddon.Addon().getSetting('lang')
+        bio = xbmcaddon.Addon().getSetting('bio')
+        if bio == 'theaudiodb' and 'theaudiodb' in details:
+            if lang in details['theaudiodb']:
+                result['biography'] = details['theaudiodb'][lang]
+            elif 'biographyEN' in details['theaudiodb']:
+                result['biography'] = details['theaudiodb']['biographyEN']
+        elif bio == 'discogs' and 'discogs' in details:
+            result['biography'] = details['discogs']['biography']
+        discog = xbmcaddon.Addon().getSetting('discog')
+        if (discog in details) and ('albums' in details[discog]):
+            result['albums'] = details[discog]['albums']
+        genre = xbmcaddon.Addon().getSetting('genre')
+        if (genre in details) and ('genre' in details[genre]):
+            result['genre'] = details[genre]['genre']
+        style = xbmcaddon.Addon().getSetting('style')
+        if (style in details) and ('styles' in details[style]):
+            result['styles'] = details[style]['styles']
+        mood = xbmcaddon.Addon().getSetting('mood')
+        if (mood in details) and ('moods' in details[mood]):
+            result['moods'] = details[mood]['moods']
+        return result
+
+    def return_search(self, data):
+        for item in data:
+            listitem = xbmcgui.ListItem(item['artist'], thumbnailImage=item['thumb'], offscreen=True)
+            listitem.setProperty('artist.genre', item['genre'])
+            listitem.setProperty('artist.born', item['born'])
+            listitem.setProperty('relevance', item['relevance'])
```

Comment:
Can all the known properties be set here not just genre, born and relevence, especially "disambiguation" when result is from MusicBrainz. When lookup is by name and many artists found a dialog is displayed to let the user choose which is the artist. Currently this will show genre and born as well as name, but usually ends up as just a list of names (most the same) which is not helpful. I'll need to change core to build better strings from the artist data, but this data also needs to be passed back from the scraper

Reply:
thx, slipped my mind to add those properties there as well.